media/src/main/java/com/tealium/media/Media.kt

```diff
@@ -0,0 +1,172 @@
+package com.tealium.media
+
+import com.tealium.core.*
+import com.tealium.dispatcher.TealiumEvent
+import com.tealium.media.segments.Ad
+import com.tealium.media.segments.AdBreak
+import com.tealium.media.segments.Chapter
+
+
+class Media(private val tealiumContext: TealiumContext,
+            var currentSession: Session? = null) : Module {
```

Comment:
would probably suggest making this a regular param, and declaring the property further down.
can have a private setter that way and is therefore only interacted with by the specified methods - i.e. a user can't overwrite it without calling `startSession` or `endSession` etc